File: Midterm/Question01.cpp
#include <iostream>
using namespace std;

int main()
{
    int n;
    cout << "Enter Height of Triangle: "; // Input for triangle height
    cin >> n;

    // Check for valid input
    if (n < 0)
    {
        cout << "Invalid Input!" << endl;
        return 1; // Exit program with error code
    }

    // First Triangle
    cout << "\nFirst Triangle:" << endl;
    for (int i = 1; i <= n; i++)
    {
        for (int j = 1; j <= i; j++)
        {
            cout << "* ";
        }
        cout << endl;
    }

    // Second Triangle
    cout << "\nSecond Triangle:" << endl;
    for (int i = 1; i <= n; i++)
    {
        for (int j = 1; j <= n - i + 1; j++)
        {
            cout << "* ";
        }
        cout << endl;
    }

    // Third Triangle
    cout << "\nThird Triangle:" << endl;
    for (int i = 1; i <= n; i++)
    {
        for (int k = 1; k <= i; k++)
        {
            cout << " "; // Print spaces for alignment
        }
        for (int j = 1; j <= n - i + 1; j++)
        {
            cout << "*";
        }
        cout << endl;
    }

    // Fourth Triangle
    cout << "\nFourth Triangle:" << endl;
    for (int i = 1; i <= n; i++)
    {
        for (int k = 1; k <= n - i + 1; k++)
        {
            cout << " "; // Print spaces for alignment
        }
        for (int j = 1; j <= i; j++)
        {
            cout << "*";
        }
        cout << endl;
    }

    return 0; // Exit program successfully
}

File: Python/Midterm/Question01.py
def main():
    n = int(input("Enter Height of Triangle: "))

    if n < 0:
        print("Invalid Input!")
        return 1

    print("\nFirst Triangle:")
    for i in range(1, n + 1):
        for j in range(1, i + 1):
            print("* ", end="")
        print()

    # Second Triangle
    print("\nSecond Triangle:")
    for i in range(1, n + 1):
        for j in range(1, n - i + 2):
            print("* ", end="")
        print()

    # Third Triangle
    print("\nThird Triangle:")
    for i in range(1, n + 1):
        for k in range(1, i + 1):
            print(" ", end="")
        for j in range(1, n - i + 2):
            print("*", end="")
        print()

    # Fourth Triangle
    print("\nFourth Triangle:")
    for i in range(1, n + 1):
        for k in range(1, n - i + 2):
            print(" ", end="")
        for j in range(1, i + 1):
            print("*", end="")
        print()

    return 0


if __name__ == "__main__":
    main()
